### PR TITLE
Complete CMake build for 5.8 release

### PR DESCRIPTION
**Pull request by compnerd**
Update CMake rules for 5.8 release to allow us to build a complete package.

### Motivation:

SPM cannot yet handle the static/dynamic linking aspect of building libraries.  Use CMake to build SPM properly until that has been resolved.  Because that will be more invasive, complete the CMake build for now.

### Modifications:

Add rules to get the remainder of SPM building on Windows.

### Result:

We get the remaining commands that were previously missing (e.g. `swift package-registry` and `swift package-collection`).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: Sources/PackageCollectionsSigning/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+target_link_libraries(PackageCollectionsSigning PUBLIC
+  $<$<NOT:$<PLATFORM_ID:Darwin>>:dispatch>
+  $<$<NOT:$<PLATFORM_ID:Darwin>>:Foundation>
+  Basics
+  CCryptoBoringSSL
+  Crypto
+  PackageCollectionsModel
+  TSCBasic)
+target_link_libraries(PackageCollectionsSigning PRIVATE
+  PackageCollectionsSigningLibc)
```
Comment (yim-lee) [inline Sources/PackageCollectionsSigning/CMakeLists.txt:46]: Do we need to link `Security` framework?

```
target_link_options(PackageCollectionsSigning PRIVATE
    "$<$<PLATFORM_ID:Darwin>:SHELL:-Xlinker -framework -Xlinker Security>")
```

Comment (compnerd) [reply Sources/PackageCollectionsSigning/CMakeLists.txt:46]: Ah, yes, we should link against Security on Darwin.

**Comment by tomerd** [comment]
@compnerd @neonichu remind me how far we are from being able to do a minimal bootstrap on windows? if we are _far_ I would be okay with taking this to 5.8, but I would not want to do the same in main and think we should make it an explicit goal to use minimal bootstrap in 5.9.

thoughts?

**Comment by compnerd** [comment]
@tomerd - we need to get the static/dynamic linking corrected in SPM, add support to prevent internal static linking to targets when dynamic linking is requested, add support to build from build-trees, and disable internal package resolution at the very least.  I think that we are pretty far away (we are talking on the order of kLoCs of changes potentially).

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
I agree with @compnerd, we're pretty far away, but I agree that we should make it a goal for 5.9

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,6 +57,7 @@ if(FIND_PM_DEPS)
   find_package(ArgumentParser CONFIG REQUIRED)
   find_package(SwiftDriver CONFIG REQUIRED)
   find_package(SwiftCollections CONFIG REQUIRED)
+  find_package(SwiftCrypto CONFIG REQUIRED)
```
[the remark this answers is not in the document]
Comment (compnerd) [reply CMakeLists.txt:60]: Sure, I can do that.

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

**Pull request merged**